core/classes/Queue/Queue.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Queue management class
+ *
+ * @package NamelessMC\Queue
+ * @author Samerton
+ * @version 2.1.0
+ * @license MIT
+ */
+class Queue {
+    /**
+     * Schedule a task
+     *
+     * @param Task $task
+     * @return bool Whether the task was scheduled successfully or not
+     */
+    public static function schedule(Task $task): bool {
+        if (!$task->getTask()) {
+            return false;
+        }
+
+        $db = DB::getInstance();
+
+        $db->query(
+            <<<SQL
+            INSERT INTO nl2_queue
+                (
+                 module_id,
+                 task,
+                 name,
+                 data,
+                 scheduled_at,
+                 scheduled_for,
+                 fragment,
+                 fragment_total,
+                 user_id,
```

Comment:
Struggling to grasp how a task could have a `user_id` when it seems like they will be enqueued and executed by modules?